contracts/TurnupSharesV4.sol

```diff
@@ -307,7 +325,14 @@ contract TurnupSharesV4 is Initializable, OwnableUpgradeable {
       _sendBuyFunds(protocolFee, subjectFee, sharesSubject);
     }
 
-    emit Trade(_msgSender(), sharesSubject, true, amount, price, protocolFee, subjectFee, supply + amount, subjectType);
+    emit Trade(_msgSender(), sharesSubject, true, amount, price, supply + amount, subjectType);
+    // It returns the excess sent by the user if any
+    return (true, expectedPrice - price - protocolFee - subjectFee);
+  }
+
+  function _sendFundsBackIfUnused(uint256 amount) internal {
+    (bool success, ) = _msgSender().call{value: amount}("");
```

Comment:
If msg.sender is a contract account, is there a risk of reentrance?

Reply:
I will fix the reentrancy issues in a following PR